tools/generate-changelog.py

```diff
@@ -0,0 +1,165 @@
+    try:
+        changelog = get_changelog_from_file(docs_d, pkg)
+    except Exception:
+        changelog = get_changelog_from_url(pkg, new_v)
+
+    source_pkg = changelog[0:changelog.find(' ')]
+
+    chl = debian.changelog.Changelog(changelog)
+    old_deb_v = debian.debian_support.Version(old_v)
+    for version in chl.get_versions():
+        vc = debian.debian_support.version_compare(old_deb_v, version)
+        if vc >= 0:
+            break
+
+    # Get the changelog chunk since the version older or equal to old_v
+    change_chunk = ''
+    old_change_start = source_pkg + ' (' + version.__str__() + ')'
+    for line in changelog.splitlines():
+        if line.startswith(old_change_start):
+            break
+        if line == '':
+            change_chunk += '\n'
+        else:
+            change_chunk += indent + line + '\n'
+
+    return source_pkg, change_chunk
+
+
+# Returns the changes related to primed packages between two manifests
+# old_manifest_p: path to old manifest
+# new_manifest_p: path to newer manifest
+# docs_d: directory with docs from debian packages
+def compare_manifests(old_manifest_p, new_manifest_p, docs_d):
+    old_packages = packages_from_manifest(old_manifest_p)
+    new_packages = packages_from_manifest(new_manifest_p)
+    changes = ''
+
+    src_pkgs = {}
+    SrcPkgData = namedtuple('SrcPkgData', 'old_v new_v changes debs')
+    for pkg, new_v in sorted(new_packages.items()):
+        try:
+            old_v = old_packages[pkg]
+            if old_v != new_v:
+                src, pkg_change = get_changes_for_version(docs_d, pkg, old_v,
+                                                          new_v, '  ')
+                if src not in src_pkgs:
+                    src_pkgs[src] = SrcPkgData(old_v, new_v, pkg_change, [pkg])
+                else:
+                    src_pkgs[src].debs.append(pkg)
+        except KeyError:
+            changes += pkg + ' (' + new_v + '): new primed package\n\n'
+
+    for src_pkg, pkg_data in sorted(src_pkgs.items()):
+        changes += ', '.join(pkg_data.debs)
+        changes += ' (built from ' + src_pkg + ') updated from '
+        changes += pkg_data.old_v + ' to ' + pkg_data.new_v + ':\n\n'
+        changes += pkg_data.changes
+
+    for pkg, old_v in sorted(old_packages.items()):
+        if pkg not in new_packages:
+            changes += pkg + ': not primed anymore\n\n'
+
+    return changes
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Manifest changelog generator")
+
+    parser.add_argument('old-manifest', help='Path to the manifest of the previous snap')
+    parser.add_argument('new-manifest', help='Path to the manifest of the new snap')
+    parser.add_argument('docs-dir', help='Path to the usr/share/doc directory in the rootfs of the new snap')
+    args = parser.parse_args()
+
+    old_manifest = getattr(args, 'old-manifest')
+    new_manifest = getattr(args, 'new-manifest')
+    docs_dir = getattr(args, 'docs-dir')
```

Comment:
If you really want to use '-' in the help, maybe you should consider using `metavar`. For example `parser.add_argument('old', metavar='old-manifest')`. And then you can use `args.old`, instead of `old_manifest = getattr(args, 'old-manifest')`. (or even use `_` in the field if you like long names).

Reply:
updated this, did not know about `metavar`, nice!